src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private int threshold;
+
+    public static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.table = (Node<K, V>[]) new Node[capacity];
+        this.size = 0;
+        this.threshold = (int) (capacity * LOAD_FACTOR);
+    }
+
+    private int hash(K key) {
+        if (key == null) {
+            return 0;
+        }
+        int hashCode = key.hashCode();
+        return hashCode == Integer.MIN_VALUE ? 0 : Math.abs(hashCode);
```

Comment:
Consider using `Math.abs(key.hashCode())` instead of `key.hashCode() & Integer.MAX_VALUE` to avoid issues with `Integer.MIN_VALUE`. This ensures that the hash function handles negative hash codes correctly.